zerver/views/push_notifications.py

```diff
@@ -254,3 +259,46 @@ def self_hosting_registration_transfer_challenge_verify(
     verification_secret = data["verification_secret"]
 
     return json_success(request, data={"verification_secret": verification_secret})
+
+
+@human_users_only
+@typed_endpoint
+def register_push_device(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    token_kind: Annotated[str, check_string_in_validator(PushDevice.TokenKind.values)],
+    push_account_id: Json[int],
+    push_public_key: str,
+    # Key used to encrypt 'encrypted_push_registration'.
```

Comment:
Let's be specific: "Key that the client claims was used to encrypt ...".

It's important to be clear about what we know / is safe to assume.

Reply:
Comment updated.